infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -122,6 +129,8 @@ private void batchWriteProcessor() {
                         streamLog.append(currOp.getLogAddress(), currOp.getLogData());
                         currOp.setException(null);
                         res.add(currOp);
+                        long globalAddress = currOp.getLogAddress().getAddress();
```

Comment:
I don't think this is the best place to maintain this. In the LogUnitServer you can have something like lastGlobalAddress = Math.max(lastGlobalAddress, currentAddress)
in the write handler method.